### PR TITLE
Support Configuring HTTP Log Detail Level with Environment Configuration

### LINKED ISSUE #20126: Enable HTTP Log Detail Level to be Configured via Environment Configurations
Currently, the default `HttpLogDetailLevel` used by `HttpLogOptions` defaults to `HttpLogDetailLevel.NONE`. To enable logging customers must modify their code to pass a log level to enable HTTP logging but this goes against general goals of logging where it should configurable without code changes. Therefore, an environment configuration should be introduced to change this, enabling changes without code modification. To align with `AZURE_LOG_LEVEL` the environment configuration key should be `AZURE_HTTP_LOG_DETAIL_LEVEL` and to maintain runtime backwards compatibility it should default to `NONE`.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #20126 

This PR enables the ability to configure the default `HttpLogDetailLevel` used in `HttpLogOptions` with an environment configuration. With this customers are able to enable HTTP request and response logging without making modifications to their code.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java [hunk 1]
```diff
@@ -34,6 +39,36 @@
      */
     BODY_AND_HEADERS;
 
+    static final String BASIC_VALUE = "basic";
+    static final String HEADERS_VALUE = "headers";
+    static final String BODY_VALUE = "body";
+    static final String BODY_AND_HEADERS_VALUE = "body_and_headers";
+    static final String BODYANDHEADERS_VALUE = "bodyandheaders";
```
Comment (JonathanGiles) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java:46]: I don't know that I would offer two variants?

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java:46]: I went with both given the underscores in the enum to allow for both a more word like and a stricter enum like case.

**Review by JonathanGiles — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java [hunk 2]
```diff
@@ -34,6 +39,36 @@
...
+
+    static final HttpLogDetailLevel ENVIRONMENT_HTTP_LOG_DETAIL_LEVEL = fromConfiguration(getGlobalConfiguration());
+
+    static HttpLogDetailLevel fromConfiguration(Configuration configuration) {
+        String detailLevel = configuration.get(Configuration.PROPERTY_AZURE_HTTP_LOG_DETAIL_LEVEL);
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java:51]:
```suggestion
        String detailLevel = configuration.get(Configuration.PROPERTY_AZURE_HTTP_LOG_DETAIL_LEVEL, "none");
```

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java [hunk 3]
```diff
@@ -34,6 +39,36 @@
...
+
+        HttpLogDetailLevel logDetailLevel;
+        if (CoreUtils.isNullOrEmpty(detailLevel)) {
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java:54]: This can be removed if the above change is made.

**Review by srnagar — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java]: Done

File: sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLogDetailLevel.java]: Done

**Pull request merged**